tools/SRS_tools/comparison_div.r

```diff
@@ -0,0 +1,191 @@
+## read selected features from dimensionality reduction
+selected_features <- read.table(sel_pc)[[1]]
+## path for selected components
+
+# location of the directory where shapefiles used for validation are saved
+dir.create("VectorDir")
+unzip(plots_zip, exdir = "VectorDir")
+
+# list vector data
+path_vector <- biodivMapR::list_shp("VectorDir")
+name_vector <- tools::file_path_sans_ext(basename(path_vector))
+
+# location of the spectral species raster needed for validation
+path_spectralspecies <- kmeans_info$SpectralSpecies
+# get diversity indicators corresponding to shapefiles (no partitioning of spectral dibversity based on field plots so far...)
+
+biodiv_indicators <- biodivMapR::diversity_from_plots(Raster_SpectralSpecies = path_spectralspecies, Plots = path_vector, nbclusters = nbclusters, Raster_Functional = pca_files, Selected_Features = selected_features)
+
+shannon_rs <- c(biodiv_indicators$Shannon)[[1]]
+fric <- c(biodiv_indicators$FunctionalDiversity$FRic)
+feve <- c(biodiv_indicators$FunctionalDiversity$FEve)
+fdiv <- c(biodiv_indicators$FunctionalDiversity$FDiv)
+# if no name for plots
+biodiv_indicators$Name_Plot <- seq(1, length(biodiv_indicators$Shannon[[1]]), by = 1)
+
+
+####################################################
+# write RS indicators                              #
+####################################################
+# write a table for Shannon index
+
+# write a table for all spectral diversity indices corresponding to alpha diversity
+results <- data.frame(name_vector, biodiv_indicators$Richness, biodiv_indicators$Fisher,
+                      biodiv_indicators$Shannon, biodiv_indicators$Simpson,
+                      biodiv_indicators$FunctionalDiversity$FRic,
+                      biodiv_indicators$FunctionalDiversity$FEve,
+                      biodiv_indicators$FunctionalDiversity$FDiv)
+
+names(results) <- c("ID_Plot", "Species_Richness", "Fisher", "Shannon", "Simpson", "fric", "feve", "fdiv")
+write.table(results, file = "Diversity.tabular", sep = "\t", dec = ".", na = " ", row.names = FALSE, col.names = TRUE, quote = FALSE)
+
+if (choice == "Y") {
+# write a table for Bray Curtis dissimilarity
+bc_mean <- biodiv_indicators$BCdiss
+bray_curtis <- data.frame(name_vector, bc_mean)
+colnames(bray_curtis) <- c("ID_Plot", bray_curtis[, 1])
+write.table(bray_curtis, file = "BrayCurtis.tabular", sep = "\t", dec = ".", na = " ", row.names = FALSE, col.names = TRUE, quote = FALSE)
+
+####################################################
+# illustrate results
+####################################################
+# apply ordination using PCoA (same as done for map_beta_div)
+
+mat_bc_dist <- as.dist(bc_mean, diag = FALSE, upper = FALSE)
+betapco <- labdsv::pco(mat_bc_dist, k = 3)
+
+# assign a type of vegetation to each plot, assuming that the type of vegetation
+# is defined by the name of the shapefile
+
+nbsamples <- shpname <- c()
+for (i in 1:length(path_vector)) {
```

Comment:
```suggestion
for (i in 1:seq_along(path_vector)) {
```

Reply:
I tried with this change but the tool does not work as well if I put seq_along()

Reply:
So perhaps it will be necessary to skip the linting of this file.

Reply:
In the pr.yaml ? I tried that too but it seems to not be working I wrote that line 202 ````lnt <- lint_dir(repo, relative_path=T, linters=linters, exclusions=list("tools/SRS_tools/comparison_div.r"=c(153, 160)))````

Reply:
According to the source code, it seems to be correct https://github.com/r-lib/lintr/blob/main/R/lint.R#L135; could you try to exclude the whole file? 

`lnt <- lint_dir(repo, relative_path=T, linters=linters, exclusions=list("tools/SRS_tools/comparison_div.r"))`